### PR TITLE
Ability to decide whether active or newly triggered (runaway) Pierre is despawned

### PR DESCRIPTION
**Pull request by Rapora9**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

If there is a runaway Pierre active and the player tries to trigger another runaway Pierre, the new Pierre will be despawned immediately while the old one remains. **With this change**, the player can change this behaviour so that old Pierre will be despawned and new Pierre remains. This way the player gets to face all instances of Pierre even if they did not make the old Pierre run before trying to trigger a new one.
...

### CODE REVIEW

File: src/game/objects/ai/pierre.c [hunk 1]
```diff
@@ -62,13 +63,30 @@ void Pierre_Control(int16_t item_num)
...
+                if (!(item->flags & IF_ONESHOT)) {
+                    KillItem(item_num);
+                }
+            } else {
+                KillItem(m_PierreItemNum);
+                m_PierreItemNum = item_num;
+            }
+        }
+    } else {
+
```
Comment (walkawayy) [inline src/game/objects/ai/pierre.c:81]: You can remove this blank line.

**Review by walkawayy — CHANGES_REQUESTED** [review]
You need to add something like this to `Tomb1Main.json5` so the user can enable/disable this setting.
```
// Fixes Pierre blah blah blah
"fix_pierre_spawn": true,
```

Can you also open an issue to describe the original game's Pierre behavior and why this is an improvement? Your videos you posted in Discord showing the difference would be useful to include in the issue.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 - added a cheat to increase the game speed (#135)
 - added a matrix stack overflow error check and message if GetRoomBounds runs infinitely (#506)
 - added ability to turn off trex collision (#437)
+- added the ability to make new Pierre replace an old active (runaway) Pierre
```
Comment (walkawayy) [inline CHANGELOG.md:8]: Also can you update this entry to include the issue number when that's opened? Also this line can probably be added to the README too (without the issue number, check out the other entries in it)

File: src/game/objects/ai/pierre.c (hunk 1 above)
Comment (Rapora9) [reply src/game/objects/ai/pierre.c]: issued opened #532. Blank line removed.

File: CHANGELOG.md (hunk 2 above)
Comment (Rapora9) [reply CHANGELOG.md]: Changelog and readme updated

**Comment by Richard-L** [comment]
Have build?

I need to learn how to make them myself at some point

**Comment by rr-** [comment]
> Have build?
> 
> I need to learn how to make them myself at some point

https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#compiling-on-windows

**Comment by walkawayy** [comment]
Here is the build @Richard-L 

[Tomb1Main-2.8-5-gf6a28ec_pierre.zip](https://github.com/rr-/Tomb1Main/files/8622281/Tomb1Main-2.8-5-gf6a28ec_pierre.zip)

**Comment by rr-** [comment]
Any updates on testing?
I think this can be merged, no?

**Comment by Richard-L** [comment]
Hmm sorry not really. I spoke about it some more with @Rapora9 over DM and the topic honestly escaped me. I cannot see how a non-TRLE player like me would be affected by this, and I think we agree that speedrun tactics around Pierre should not be broken.

**Comment by Rapora9** [comment]
> Any updates on testing? I think this can be merged, no?

Yes, I think it's good to go. I haven't spotted any issues anymore.

**Comment by rr-** [comment]
Merged manually to the develop branch. Thanks @Rapora9

**Pull request closed without merging**